ballerina/stream_implementor.bal

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+class ListStream {
+    private Stream[] currentEntries = [];
+    private int index = 0;
+    private final http:Client httpClient;
+    private final string accessKeyId;
+    private final string secretAccessKey;
+    private final string region;
+    private final string awsHost;
+    private final string uri = SLASH;
+    private string? lastEvaluatedStreamArn;
+    private ListStreamsInput listStreamInput;
+
+    isolated function init(http:Client httpClient, string host, string accessKey, string secretKey, string region, ListStreamsInput streamInput)
+                           returns error? {
+        self.httpClient = httpClient;
+        self.accessKeyId = accessKey;
+        self.secretAccessKey = secretKey;
+        self.region = region;
+        self.awsHost = AWS_STREAMS_SERVICE + DOT + self.region + DOT + AWS_HOST;
+        self.lastEvaluatedStreamArn = null;
+        self.listStreamInput = streamInput;
+        self.currentEntries = check self.fetchStreams();
+    }
+
+    public isolated function next() returns record {| Stream value; |}|error? {
+        if (self.index < self.currentEntries.length()) {
```

Comment:
```suggestion
        if self.index < self.currentEntries.length() {
```

Reply:
please check other places

Reply:
Fixed in [9a61dc7](https://github.com/ballerina-platform/module-ballerinax-aws.dynamodbstreams/pull/2/commits/9a61dc79f52588cd3098b3a576772e734b616e93)